octue/utils/threads.py

```diff
@@ -0,0 +1,44 @@
+import time
+from subprocess import PIPE, STDOUT, CalledProcessError, Popen
+from threading import Thread, Timer
+
+
+class RepeatingTimer(Timer):
+    """A repeating version of the `threading.Timer` class."""
+
+    def run(self):
+        while not self.finished.is_set():
+            time.sleep(self.interval)
+            self.function(*self.args, **self.kwargs)
+
+
+def run_subprocess_and_log_stdout_and_stderr(command, logger, log_level="info", *args, **kwargs):
```

Comment:
How about `run_logged_subprocess`, which is sufficiently descriptive but which we won't have to change if there's future engineering work around how and what we log in subprocesses

Reply:
Yeah that's a better name and I might as well given that I've moved it in this PR